### PR TITLE
auth-cart-wishlist

### PR DESCRIPTION
**Pull request by anshulraheja**
1. Auth Functionality - Signup and login 
2. Cart Functionality - Add to cart, remove from cart, increase/ decrease qty
3. Wishlist Functionality - add or remove to wishlist 
Demo: https://624c8768cfd7a7198b1d717d--legendary-lokum-2ff412.netlify.app/

### CODE REVIEW

File: src/components/Header/Header.js [hunk 1]
```diff
@@ -28,18 +33,25 @@ const Header = () => {
...
                     </Link>
                     <Link to="/cart" className="badge-wrapper">
                         <i className="fas fa-shopping-bag"></i>
-                        <span className="badge notification-badge">2</span>
-                    </Link>
-                    <Link to="/signin" className="btn-login">
-                        Sign in
+                        {cartState.cart.length > 0 && <span className="badge notification-badge">{cartState.cart.length}</span>}
                     </Link>
+                    {auth.isLoggedIn === true ?
```
Comment (sanjitsarkar) [inline src/components/Header/Header.js:42]: No need to add ` auth.isLoggedIn === true` just write `auth.isLoggedIn`

File: src/components/ProductCard/HorizontalProductCard.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import './ProductCard.css'
+import { useWishlist } from '../../context/wishlist-context.js'
+import { useCart } from '../../context/cart-context';
+const HorizontalProductCard = ({ item }) => {
+    const { title, price, image, categoryName, subCategory, rating } = item;
+    // const { wishlist, toggleWishlist } = useWishlist();
```
Comment (sanjitsarkar) [inline src/components/ProductCard/HorizontalProductCard.js]: remove commented code

File: src/components/ProductCard/ProductCard.css [hunk 3]
```diff
@@ -79,4 +84,10 @@
     color: #f67e09;
```
Comment (sanjitsarkar) [inline src/components/ProductCard/ProductCard.css:84]: use variable for color

File: src/components/ProductCard/ProductCard.css [hunk 4]
```diff
@@ -43,6 +43,11 @@
     background-color: var(--COLOR-BLACK);
     color: var(--COLOR-WHITE);
 }
+.btn-goToBag{
+    text-align: center;
+    background-color: var(--COLOR-ADDTOCART);
```
Comment (sanjitsarkar) [inline src/components/ProductCard/ProductCard.css:48]: use lowerCase for naming css variable

File: src/context/auth-context.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                    "email": email,
+                    "password": password,
+                    "confirmpassword": confirmpassword,
+                    "firstName": name,
+                    "terms": terms
+                });
+                authDispatcher({ type: "LOGGED_IN", payload: response.data });
+                navigate("/");
+            } catch (error) {
+                alert(error);
```
Comment (sanjitsarkar) [inline src/context/auth-context.js:63]: remove alert in production

File: src/context/cart-context.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        {
+            token &&
+                (async () => {
+                    try {
+                        const { data: { cart } } = await axios("api/user/cart", {
+                            headers: {
+                                authorization: token,
+                            }
+                        })
+                        console.log("cart useEffect:", cart)
```
Comment (sanjitsarkar) [inline src/context/cart-context.js]: remove console log

File: src/pages/SignupPage/SignupPage.css [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+.login-form .login-header{
+    font-size: 2rem;
+    font-weight: 500;
+    text-align: center;
+}
+.login-form label{
+    display: inline-block;
+    font-weight: 500;
+}
+.login-form input[type="email"], input[type="password"], input[type="name"], input[type="confirmpassowrd"]{
```
Comment (sanjitsarkar) [inline src/pages/SignupPage/SignupPage.css]: there is no any input type attribute  confirmpassowrd

**Review by sanjitsarkar — APPROVED** [review]
Rest looks good

File: src/context/auth-context.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        e.preventDefault();
+        const { email, password } = userInfo;
+        if (email == "" || password == "") {
+            alert("Fields can't be empty");
+            return;
+        }
+        else {
+
+            try {
+                const response = await axios.post("/api/auth/login", JSON.stringify({
```
Comment (rahulyadav139) [inline src/context/auth-context.js:84]: I don't think you need to use the JSON method to specify body type. Axios already does that in the background. Try that.

File: src/pages/LoginPage/LoginPage.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        password: "",
+    });
+
+    const { loginHandler, auth } = useAuth();
+
+    const inputHandler = (e) => {
+        if (e.target.type === "checkbox") {
+            setloginData((loginData) => ({
+                ...loginData,
+                [e.target.name]: e.target.checked,
```
Comment (rahulyadav139) [inline src/pages/LoginPage/LoginPage.js:19]: This is great. you are using the input name as the object key. great work.

**Review by rahulyadav139 — APPROVED** [review]

**Pull request merged**